face/unix-stream-transport.go

```diff
@@ -117,20 +117,32 @@ func (t *UnixStreamTransport) runReceive() {
 		}
 
 		// Determine whether valid packet received
-		for startPos > 0 {
-			_, _, tlvSize, err := tlv.DecodeTypeLength(recvBuf)
+		tlvPos := 0
+		for {
+			if tlvPos >= startPos {
+				startPos = 0
+				break
+			}
+
+			_, _, tlvSize, err := tlv.DecodeTypeLength(recvBuf[tlvPos:])
 			if err != nil {
 				core.LogInfo(t, "Unable to process received packet: ", err)
 				startPos = 0
 				break
-			} else if startPos >= tlvSize {
+			} else if startPos >= tlvPos+tlvSize {
 				// Packet was successfully received, send up to link service
-				t.linkService.handleIncomingFrame(recvBuf[:tlvSize])
-				startPos -= tlvSize
-				if startPos > 0 {
-					copy(recvBuf, recvBuf[tlvSize:])
-				}
+				t.linkService.handleIncomingFrame(recvBuf[tlvPos : tlvPos+tlvSize])
+				tlvPos += tlvSize
 			} else {
+				if tlvPos > 0 {
+					if startPos > tlvPos {
+						// Move remaining data to beginning of buffer
+						copy(recvBuf, recvBuf[tlvPos:startPos])
+						startPos -= tlvPos
+					} else {
+						startPos = 0
```

Comment:
Want to confirm when this line is executed.
I don't think `tlvPos == startPos` will lead the program here, since you have already broken the loop at L122-125.
If it hits when `tlvPos == startPos`, `startPos -= tlvPos` should also work well. And then, my suggestion is to move `startPos -= tlvPos` outside of the `if startPos > tlvPos` branch.

Reply:
Good catch. I'll update this.

Reply:
Done

Reply:
Thanks